object/encryptionkey.go

```diff
@@ -0,0 +1,270 @@
+package object
+
+import (
+	"bytes"
+	"crypto/cipher"
+	"encoding"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+
+	"go.sia.tech/renterd/internal/utils"
+	"golang.org/x/crypto/chacha20"
+	"lukechampine.com/frand"
+)
+
+type EncryptionKeyInterface interface {
+	encoding.BinaryMarshaler
+	encoding.BinaryUnmarshaler
+	encoding.TextMarshaler
+	encoding.TextUnmarshaler
+	fmt.Stringer
+}
+
+var NoOpKey = EncryptionKey{
+	entropy: new([32]byte),
+	keyType: EncryptionKeyTypeBasic,
+}
+
+type EncryptionKeyType int
+
+const (
+	EncryptionKeyTypeBasic = EncryptionKeyType(iota + 1)
+	EncryptionKeyTypeSalted
+)
+
+// A EncryptionKey can encrypt and decrypt messages.
+type EncryptionKey struct {
+	entropy *[32]byte         `json:"-"`
```

Comment:
both fields are not exposed so they're not json encoded by default, making the annotation redundant?